Sources/StreamChat/Utils/LazyCachedMapCollection.swift

```diff
@@ -29,7 +29,16 @@ public struct LazyCachedMapCollection<Element>: RandomAccessCollection {
         map: @escaping (SourceElement) -> Element
     ) where Collection.Element == SourceElement, Collection.Index == Index {
         generator = { map(source[$0]) }
-        cache = .init(capacity: source.count)
+
+        /// This is just an internal test to see how we behave when the DB models are immediately mapped instead of being lazily mapped
+        if StreamRuntimeCheck._isLazyMappingEnabled {
+            cache = .init(capacity: source.count)
+        } else {
+            if Thread.isMainThread {
+                log.debug("This should not be happening on the Main Thread")
+            }
+            cache = .init(elements: source.map(map))
+        }
```

Comment:
If there's no drop in performance, perhaps, we can just go for it and evaluate models immediately? If that's the case, can we start exposing a plain array of channels/messages by controllers?

Reply:
This is what I want to work on the following PR. For now, I want to keep it easy, and fully validate the assumptions